filebeat/input/journald/pkg/journalfield/matcher.go

```diff
@@ -36,12 +36,26 @@ type MatcherBuilder struct {
 	Conversions map[string]Conversion
 }
 
+// IncludeMatches stores the advanced matching configuratio
+// provided by the user.
+type IncludeMatches struct {
+	Matches []Matcher        `config:"match"`
+	AND     []IncludeMatches `config:"and"`
+	OR      []IncludeMatches `config:"or"`
+}
+
 type journal interface {
 	AddMatch(string) error
 	AddDisjunction() error
+	AddConjunction() error
 }
 
-var defaultBuilder = MatcherBuilder{Conversions: journaldEventFields}
+var (
+	defaultBuilder = MatcherBuilder{Conversions: journaldEventFields}
+	coreDumpMsgID  = MustBuildMatcher("message_id=fc2e22bc6ee647b6b90729ab34a250b1")
+	journaldUID    = MustBuildMatcher("journald.uid=0")
+	journaldPID    = MustBuildMatcher("journald.pid=1")
```

Comment:
I like the comment you added to `ApplyUnitMatchers`, however I think it could also be added here, so they don't look like magic strings to the reader.

Anyway, it's not a blocker ;)